.github/workflows/deploy.yml

```diff
@@ -0,0 +1,24 @@
+name: Deploy documentation
+on:
+  push:
+    branches:
+      - master
+permissions:
+  contents: write
+jobs:
+  build-and-deploy:
+    concurrency: ci-${{ github.ref }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v3
+
+      - name: Install and Build 🔧
+        run: |
+          npm ci
+          npm run build:doc
+
+      - name: Deploy 🚀
+        uses: JamesIves/github-pages-deploy-action@v4
+        with:
+          folder: dist/doc
```

Comment:
```suggestion
name: Deploy documentation

on:
  push:
    branches:
      - master

permissions:
  contents: write
  
jobs:
  build-and-deploy:
    concurrency: ci-${{ github.ref }}
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Read package.json node and npm engines version
        uses: skjnldsv/read-package-engines-version-actions@v2
        id: versions
        with:
          fallbackNode: '^14'
          fallbackNpm: '^7'

      - name: Set up node ${{ steps.versions.outputs.nodeVersion }}
        uses: actions/setup-node@v3
        with:
          node-version: ${{ steps.versions.outputs.nodeVersion }}

      - name: Set up npm ${{ steps.versions.outputs.npmVersion }}
        run: npm i -g npm@"${{ steps.versions.outputs.npmVersion }}"

      - name: Install dependencies & build
        run: |
          npm ci
          npm run build:doc --if-present

      - name: Deploy
        uses: JamesIves/github-pages-deploy-action@v4
        with:
          branch: gh-pages
          folder: dist/doc
```

Reply:
`branch: gh-pages` is not needed as that is the default, but I think adding it does not hurt

Reply:
Ah, good to know! :)